R/filterReadsBam.R

```diff
@@ -0,0 +1,174 @@
+#'
+#' For each file in \code{infiles}, parse alignments, calculate read statistics
+#' and write the alignment to the corresponding output file from \code{outfiles}
+#' if the read passes all criteria defined by the filtering arguments.
+#' Filters are processed hierarchically: If a read does not pass a given
+#' filter, the remaining filters will not be examined and the processing
+#' continues with the next read.
+#' The filters are examined in this order: \code{minReadLength},
+#' \code{minAlignedLength}, \code{minAlignedFraction}, \code{minQscore},
+#' \code{maxFracLowConf}, \code{maxEntropy}.
+#'
+#' @param infiles Character vector with name(s) of the input bam file(s).
+#'     If \code{infiles} is a named vector, the names are used as sample names.
+#'     Otherwise, the each file will be names as \code{s1}, ..., \code{sN},
+#'     where \code{N} is the length of \code{infiles}.
+#' @param outfiles Character vector with name(s) of the output bam file(s).
+#'     Needs to have the same length as \code{infiles}.
+#' @param modbase Character scalar defining the modified base to analyze
+#'     (used by \code{maxEntropy} and \code{maxFracLowConf}).
+#' @param indexOutfiles Logical scalar. If \code{TRUE} (the default) create
+#'     a bam index file (\code{.bai} file) for each of the generated
+#'     \code{outfiles}.
+#' @param minReadLength A numeric scalar representing the smallest acceptable
+#'     read length. Reads that are shorter than this value will be filtered
+#'     out.
+#' @param minAlignedLength A numeric scalar representing the smallest acceptable
+#'     aligned length. Reads with aligned length shorter than this value will
+#'     be filtered out.
+#' @param minAlignedFraction A numeric scalar representing the smallest
+#'     acceptable aligned fraction of a read. Reads where the aligned fraction
+#'     is smaller than this value will be filtered out.
+#' @param minQscore A numeric scalar representing the smallest acceptable
+#'     read-level Qscore. Reads with Qscore below this value will be filtered
+#'     out.
+#' @param maxFracLowConf A numeric scalar representing the maximally acceptable
+#'     fraction of low-confidence modified base calls in a read. Reads with
+#'     a fraction of low confidence calls greater than this value will be
+#'     filtered out.
+#' @param maxEntropy A numeric scalar representing the largest acceptable
+#'     read-level entropy. Reads with entropy above this value will be filtered
+#'     out. A value of \code{Inf} deactivates the entropy filter.
+#' @param BPPARAM A \code{\link[BiocParallel]{BiocParallelParam}} object that
+#'     controls the number of parallel CPU threads to use for some of the steps
+#'     in \code{filterReadsBam()}. The default value is
+#'     (\code{\link[BiocParallel]{MulticoreParam}(4L, RNGseed = 42L)}).
+#' @param verbose Logical scalar. If \code{TRUE}, report on progress.
+#'
+#' @return A \code{data.frame} with one row per \code{infile} giving the numbers
```

Comment:
Mention also the side effect of generating the filtered bam files here?

Reply:
good point - added in 4d831cc7